src/core/meshcop/leader.cpp

```diff
@@ -86,8 +86,7 @@ void Leader::HandlePetition(Coap::Header &aHeader, Message &aMessage, const Ip6:
     data.mCommissionerSessionId.SetCommissionerSessionId(++mSessionId);
 
     data.mSteeringData.Init();
-    data.mSteeringData.SetLength(1);
-    data.mSteeringData.Clear();
+    data.mSteeringData.SetLength(0);
```

Comment:
Yes, OT works the correct way according to the spec. I will close this PR and ask SiLabs for the fix, thanks.